internal/platform/gin/auditlog/auditlogdriver/log.go

```diff
@@ -0,0 +1,82 @@
+// Copyright © 2020 Banzai Cloud
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package auditlogdriver
+
+import (
+	"github.com/banzaicloud/pipeline/internal/platform/gin/auditlog"
+)
+
+// LogDriverConfig configures a standard output audit log driver.
+type LogDriverConfig struct {
+	Verbosity int
+}
+
+// Logger is the fundamental interface for all log operations.
+type Logger interface {
+	// Info logs an info event.
+	Info(msg string, fields ...map[string]interface{})
+}
+
+// NewLogDriver returns a standard output audit log driver.
+func NewLogDriver(config LogDriverConfig, logger Logger) auditlog.Driver {
+	return logDriver{
+		config: config,
+		logger: logger,
+	}
+}
+
+type logDriver struct {
+	config LogDriverConfig
+	logger Logger
+}
+
+func (d logDriver) Store(entry auditlog.Entry) error {
+	if d.config.Verbosity == 0 {
+		return nil
+	}
+
+	data := make(map[string]interface{})
+
+	if d.config.Verbosity >= 1 {
+		data["timestamp"] = entry.Time
+		data["correlationID"] = entry.CorrelationID
+		data["userID"] = entry.UserID
+	}
+
+	if d.config.Verbosity >= 2 {
+		data["http.method"] = entry.HTTP.Method
+		data["http.path"] = entry.HTTP.Path
+		data["http.clientIP"] = entry.HTTP.ClientIP
+	}
+
+	if d.config.Verbosity >= 3 {
+		data["http.userAgent"] = entry.HTTP.UserAgent
+		data["http.statusCode"] = entry.HTTP.StatusCode
+	}
+
+	if d.config.Verbosity >= 4 {
+		data["http.responseTime"] = entry.HTTP.ResponseTime
+		data["http.responseSize"] = entry.HTTP.ResponseSize
+		data["http.requestBody"] = entry.HTTP.RequestBody
+
+		if len(entry.HTTP.Errors) > 0 {
+			data["http.errors"] = entry.HTTP.Errors
+		}
+	}
```

Comment:
I would consider 1,2,3 as simply metadata (and would add response time and reponse size as well) and enable it with a single flag. Then use the verbose flag to enable request (1) and response (2) body as well.

Reply:
Response body is not recorded. For development, we never needed it. For auditing, I don't see why it's relevant: the state is there, we are mostly concerned about the request part.

I would consider adding the response body as a separate proposal. Without that, if I understand correctly, you propose moving response time and size to level 3. Is that correct?

Reply:
sounds like a deal, but shouldn't we make verbosity 3 by default?

Reply:
however from auditing point of view, why wouldn't response body also be helpful, i.e. to see what information had been leaked in a specific case?